util/grpcerrors/grpcerrors.go

```diff
@@ -72,6 +73,23 @@ func ToGRPC(ctx context.Context, err error) error {
 	return st.Err()
 }
 
+// errorHasMoreContext checks if the original error provides more context by having
+// a different message or additional details than the Status.
+func errorHasMoreContext(err error, st *status.Status) bool {
+	if errMessage := err.Error(); len(errMessage) > len(st.Message()) {
+		// check if the longer message in errMessage is only due to
+		// prepending with the status code
+		prefix := st.Code().String() + ": "
```

Comment:
is this trying to mimic `grpcStatusError.Error()`. In that case looks like it is more complicated as some codes have special meaning. Type checking to `*grpcStatusError` could be better.